core/crypto/pedersen_hash.go

```diff
@@ -61,25 +21,6 @@ func PedersenArray(elems ...*felt.Felt) *felt.Felt {
 // [Pedersen hash]: https://docs.starknet.io/documentation/develop/Hashing/hash-functions/#pedersen_hash
 // [reference implementation]: https://github.com/starkware-libs/cairo-lang/blob/de741b92657f245a50caab99cfaef093152fd8be/src/starkware/crypto/signature/fast_pedersen_hash.py
 func Pedersen(a *felt.Felt, b *felt.Felt) *felt.Felt {
-	result := new(starkcurve.G1Jac).Set(shiftPoint)
-	result.AddAssign(processElement(a.Impl(), p0, p1))
-	result.AddAssign(processElement(b.Impl(), p2, p3))
-
-	var affine starkcurve.G1Affine
-	affine.FromJacobian(result)
-	return felt.NewFelt(&affine.X)
-}
-
-func processElement(a *fp.Element, p1 *starkcurve.G1Jac, p2 *starkcurve.G1Jac) *starkcurve.G1Jac {
-	var bigInt big.Int
-	var aBytes [32]byte
-	a.BigInt(&bigInt).FillBytes(aBytes[:])
-
-	highPart := bigInt.SetUint64(uint64(aBytes[0])) // The top nibble (bits 249-252)
-	lowPart := aBytes[1:]                           // Zero-out the top nibble (bits 249-252)
-
-	m := new(starkcurve.G1Jac).ScalarMultiplication(p2, highPart)
-	var n starkcurve.G1Jac
-	n.ScalarMultiplication(p1, bigInt.SetBytes(lowPart))
-	return m.AddAssign(&n)
+	hash := pedersenhash.Pedersen(&a.Val, &b.Val)
+	return felt.NewFelt(hash)
```

Comment:
I'm worried that these extra conversions may slow down our hash calculation a bit. Thoughts?

Reply:
I doubt it would. Can you try benchmarking it?